libs/langchain-anthropic/src/experimental/utils/tool_calling.ts

```diff
@@ -74,9 +74,18 @@ export function fixArrayXMLParameters(
     const schemaType = (schema.properties[key] as any).type;
     // Crawl for lists indistinguishable from single items
     if (schema.properties && schema.properties[key] && schemaType === "array") {
-      fixedParameters[key] = Array.isArray(xmlParameters[key])
-        ? xmlParameters[key]
-        : [xmlParameters[key]];
+      const value = xmlParameters[key];
+      if (Array.isArray(value)) {
+          fixedParameters[key] = value;
+      } else if (typeof value === 'string') {
+        if (value.startsWith("[") && value.endsWith("]")) {
```

Comment:
Some of these heuristics might not be good, but it resolved the problems I encountered

Reply:
I think looks reasonable - basically it's sometimes returning an array or object as JSON?

Reply:
I guess there's the very off chance that someone asked for a JSON string - how commonly were you seeing this and with what params?

Reply:
I specifically saw this for number arrays and there it always responded with `<key>[1,2,3]</key>` or <key>1,2,3</key>, never the proper XML way.

I was using Opus with temperature 0

Reply:
Oh right, and we expect it to be an array based on the schema anyway. Then this makes sense, thank you!